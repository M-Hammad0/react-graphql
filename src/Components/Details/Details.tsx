import React from "react";
import { useQuery, gql } from "@apollo/client";
import { useParams } from "react-router";


interface DetailsI {
  id: string;
  mission_name: string;
  launch_date_local: string;
  launch_success: boolean;
  links: {
    flickr_images: Array<String>;
    article_link: Object;
  };
}

interface DetialsData {
  launches: DetailsI[];
}

interface DetailsVars {
  slug: string
}

export const GET_LAUNCH_DETAILS = gql`
query getLaunchDetails($slug: String!){
        launches(find: {mission_name: $slug}){
          id
          mission_name
          launch_success
          details
          links {
            article_link
            flickr_images
          }
        }
  }
`;

function Details() {
  const {mission} = useParams()
  const { loading,data,error } = useQuery<DetialsData, DetailsVars>(GET_LAUNCH_DETAILS,{
    variables: {
      slug: mission
    }
  });

  if(error) return <p>{error}</p>
  if(!data && loading) return <p>loading....</p>

  return (
    <div>
      <h1>show plz</h1>
      <h1 data-testid="launch-detail" >{data?.launches.map(i => (<p key={i.id}>{i.mission_name}</p>))}</h1>
    </div>
  );
}
export default Details;
