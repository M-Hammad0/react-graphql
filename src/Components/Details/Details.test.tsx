import React from 'react'
import { render, cleanup } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import Details, { GET_LAUNCH_DETAILS } from './Details';

const mocks = [
      {
        request: {
          query: GET_LAUNCH_DETAILS,
          variables: {
                slug: "CRS-7"
          },
        },
        result: {
            data: {
              launches: [
                {
                  id: "24",
                  details: "Launch performance was nominal until an overpressure incident in the second-stage LOX tank, leading to vehicle breakup at T+150 seconds. The Dragon capsule survived the explosion but was lost upon splashdown because its software did not contain provisions for parachute deployment on launch vehicle failure.",
                  launch_success: false,
                  mission_nam: "CRS-7",
                  link: {
                    article_link: "https://spaceflightnow.com/2015/06/28/falcon-9-rocket-destroyed-in-launch-mishap/",
                    flickr_images: [
                      "https://farm1.staticflickr.com/344/19045370790_f20f29cd8d_o.jpg",
                      "https://farm1.staticflickr.com/287/18999110808_6e153fed64_o.jpg"
                    ]
                  }
                }
              ]
            }
          }
      },
    ];

afterEach(cleanup);

it('should render component',() => {
      const {asFragment} = render(
      <MockedProvider mocks={mocks} addTypename={false}>
      <Details />
      </MockedProvider>)
      expect(asFragment).toMatchSnapshot();
})

it('should render launch details',async () => {
      const {findByTestId,findByText} = render(
            <MockedProvider mocks={mocks} addTypename={false}>
            <Details />
            </MockedProvider>)
            const missionName = await findByText("CRS-7")
            expect(missionName).toBeInTheDocument();
})