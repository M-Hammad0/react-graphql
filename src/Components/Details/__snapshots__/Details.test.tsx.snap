// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render details component 1`] = `[Function]`;
